neuroanalysis/fitting.py

```diff
@@ -93,31 +93,35 @@ def make_params(self, **params):
             http://lmfit.github.io/lmfit-py/constraints.html#constraints-chapter
         """
         p = lmfit.Parameters()
+
+        #NOTE: the order of parameters can affect fit.
         for k in self.param_names:
             p.add(k)
         
-        for param,val in params.items():
+        for param, val in params.items():
+            if not isinstance(val, tuple) or len(val)<2 or len(val)>3: 
+                raise TypeError("Tuple parameter specifications must be (val, 'fixed')"
+                                    " or (val, min, max).")
+            # functions can be used as parameters to be fit.  Add any parameters that are used for 
+            # these sorts of functions as they will not be i self.param_names
             if param not in p:
                 p.add(param)
-                
-            if isinstance(val, str):
-                p[param].expr = val
-            elif np.isscalar(val):
+
+            # specify the first value in the tuple as a expression or an initial condition    
+            if isinstance(val[0], str): #if the first item in the tuple is a string
```

Comment:
I don't understand the API change here -- we need to specify expressions as `('expr',)` instead of `'expr'` ? (and same with initial values?)

Reply:
Yes, that is exactly right. Making the inputs consistent made the code much more simple to understand.